### PR TITLE
Fix ClickHouse table property documentation and examples

### PR DESCRIPTION
**Pull request by hashhar**
See https://github.com/trinodb/trino/issues/8331 for some context.

### CODE REVIEW

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 1]
```diff
@@ -90,13 +92,13 @@ Property Name               Default Value    Description
 =========================== ================ ==============================================================================================================
 ``engine``                  ``Log``          Name and parameters of the engine.
 
-``order_by``                (none)           list of columns to be the sorting key. Required if ``engine`` is ``MergeTree``.
+``order_by``                (none)           Array of columns or expressions to use as the sorting key. Required if ``engine`` is ``MergeTree``.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/clickhouse.rst:95]: Array of columns or expressions to concatenate to create the sorting key.

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 2]
```diff
@@ -90,13 +92,13 @@ Property Name               Default Value    Description
...
 
-``partition_by``            (none)           list of columns to be the partition key. Optional.
+``partition_by``            (none)           Array of columns or expressions to use as the partition key. Optional.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/clickhouse.rst:97]: Array of columns or expressions to use to use as nested partition keys.

Assuming they are not concatenated but instead created nested directories of partitions

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 3]
```diff
@@ -90,13 +92,13 @@ Property Name               Default Value    Description
...
 
-``primary_key``             (none)           list of columns to be the primary key. Optional.
+``primary_key``             (none)           Array of columns or expressions to use as the primary key. Optional.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/clickhouse.rst:99]: Array of columns or expressions to concatenate to create the primary key.

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 4]
```diff
@@ -90,13 +92,13 @@ Property Name               Default Value    Description
...
 
-``sample_by``               (none)           An expression for sampling. Optional.
+``sample_by``               (none)           An expression to use for sampling. Optional.
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/clickhouse.rst:101]: Wording is okay .. but what does that mean? If this is used for table stats .. then maybe use

An expression to use for sampling data for table statistics generation.

**Review by mosabua — CHANGES_REQUESTED** [review]

**Review by findepi — APPROVED** [review]
% Manfred's

File: docs/src/main/sphinx/connector/clickhouse.rst (hunk 4 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/clickhouse.rst:101]: It allows you to perform the equivalent of `TABLESAMPLE` in ClickHouse. See https://clickhouse.tech/docs/en/sql-reference/statements/select/sample/.

i.e. it's useful to set if someone is accessing the table from ClickHouse.

**Comment by hashhar** [comment]
AC @mosabua. PTAL.

File: docs/src/main/sphinx/connector/clickhouse.rst (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/clickhouse.rst:101]: Lets add a link to that then..

Comment (hashhar) [reply docs/src/main/sphinx/connector/clickhouse.rst]: Added. Please take a look.

**Comment by hashhar** [comment]
@mosabua PTAL added the link to https://clickhouse.tech/docs/en/sql-reference/statements/select/sample/ as requested.

**Review by mosabua — APPROVED** [review]
Looks good now. Thanks @hashhar !

**Review by Ordinant — APPROVED** [review]
Wording is fine. No way to assess technical correctness.

**Review by wgzhao — APPROVED** [review]

**Pull request merged**